build/source/dshare/get_ixname.f90

```diff
@@ -836,9 +818,135 @@ function get_ixbvar(varName)
   case('averageRoutedRunoff'           ); get_ixbvar = iLookBVAR%averageRoutedRunoff             ! routed runoff (m s-1)
   ! get to here if cannot find the variable
   case default
-   get_ixbvar = imiss
+   get_ixbvar = integerMissing
  endselect
  end function get_ixbvar
 
+ ! *********************************************************************************************************
+ ! public function get_ixVarType: get the index of the named variable type
+ ! *********************************************************************************************************
+ function get_ixVarType(varType)
+ USE var_lookup,only:iLookVarType                    ! indices of the named variable types
+ implicit none
+ ! define dummy variables
+ character(*), intent(in) :: varType                 ! variable type name
+ integer(i4b)             :: get_ixVarType          ! index of the named variable type list
+ ! get the index of the named variables
+ select case(trim(varType))
+  case('scalarv'); get_ixVarType = iLookVarType%scalarv
+  case('wLength'); get_ixVarType = iLookVarType%wLength
+  case('midSnow'); get_ixVarType = iLookVarType%midSnow
+  case('midSoil'); get_ixVarType = iLookVarType%midSoil
+  case('midToto'); get_ixVarType = iLookVarType%midToto
+  case('ifcSnow'); get_ixVarType = iLookVarType%ifcSnow
+  case('ifcSoil'); get_ixVarType = iLookVarType%ifcSoil
+  case('ifcToto'); get_ixVarType = iLookVarType%ifcToto
+  case('routing'); get_ixVarType = iLookVarType%routing
+  case('unknown'); get_ixVarType = iLookVarType%unknown
+  ! get to here if cannot find the variable
+  case default
+   get_ixVarType = integerMissing
```

Comment:
I don't think so. the others do not have this, and the warning message should be caught in the call to this function. In some other cases with these `get_ixname` functions, we want to have the option to return a "not found" answer. Perhaps that is not necessary when indexing variable types, but perhaps we should treat it with the same convention as the other `get name` type functions.